core/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -864,15 +925,37 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
     }
 
     /**
+     * Deletes a snapshot from the repository, looking up the {@link Snapshot} reference before deleting.
+     * If the snapshot is still running cancels the snapshot first and then deletes it from the repository.
```

Comment:
@ywelsch and I discussed this; because two different threads will submit cluster state update tasks to remove the current snapshot from the cluster state, and one of them would try to delete physical contents on disk (which don't exist) via `clusterStateProcessed` callback, and since we are guaranteed to have the "in-progress" snapshot removed by `removeSnapshotFromClusterState` (because the snapshot name exists check will fail), we will leave it as is.